diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/MagicNumberRule.kt

```diff
@@ -0,0 +1,100 @@
+        )
+        if (node.elementType == INTEGER_CONSTANT || node.elementType == FLOAT_CONSTANT) {
+            checkNumber(node, configuration)
+        }
+    }
+
+    @Suppress("ComplexMethod")
+    private fun checkNumber(node: ASTNode, configuration: MagicNumberConfiguration) {
+        val nodeText = node.treePrev?.let { if (it.elementType == OPERATION_REFERENCE && it.hasChildOfType(MINUS)) "-${node.text}" else node.text } ?: node.text
+        val isIgnoreNumber = configuration.ignoreNumbers.contains(nodeText)
+        val isHashFunction = node.parent({ it.elementType == FUN && it.isHashFun() }) != null &&
+                node.parents().find { it.elementType == PROPERTY } == null
+        val isPropertyDeclaration = node.parent({ it.elementType == PROPERTY && !it.isNodeFromCompanionObject() }) != null
+        val isLocalVariable = node.parent({ it.isVarProperty() && (it.psi as KtProperty).isLocal }) != null
+        val isConstant = node.parent({ it.elementType == PROPERTY && it.isConstant() }) != null
+        val isCompanionObjectProperty = node.parent({ it.elementType == PROPERTY && it.isNodeFromCompanionObject() }) != null
+        val isEnums = node.parent({ it.elementType == ENUM_ENTRY }) != null
+        val isRanges = node.treeParent.run {
+            this.elementType == BINARY_EXPRESSION &&
+                    this.findChildByType(OPERATION_REFERENCE)?.hasChildOfType(RANGE) ?: false
+        }
+        val isExtensionFunctions = node.parent({ it.elementType == FUN && (it.psi as KtFunction).isExtensionDeclaration() }) != null &&
+                node.parents().find { it.elementType == PROPERTY } == null
+        val result = listOf(isHashFunction, isPropertyDeclaration, isLocalVariable, isConstant,
+            isCompanionObjectProperty, isEnums, isRanges, isExtensionFunctions).zip(mapConfiguration.map { configuration.getParameter(it.key) })
+        if (result.any { it.first && it.first != it.second } && !isIgnoreNumber) {
+            MAGIC_NUMBER.warn(configRules, emitWarn, isFixMode, nodeText, node.startOffset, node)
+        }
+    }
+
+    private fun ASTNode.isHashFun() =
+            (this.psi as KtFunction).run {
+                this.nameIdentifier?.text == "hashCode" && this.annotationEntries.map { it.text }.contains("@Override")
+            }
+
+    /**
+     * [RuleConfiguration] for configuration
+     */
+    class MagicNumberConfiguration(config: Map<String, String>) : RuleConfiguration(config) {
+        /**
+         * List of ignore numbers
+         */
+        val ignoreNumbers = config["ignoreNumbers"]?.split(",")?.map { it.trim() } ?: ignoreNumbersList
+
+        /**
+         * @param param parameter from config
+         * @return value of parameter
+         */
+        @Suppress("UnsafeCallOnNullableType")
+        fun getParameter(param: String) = config[param]?.toBoolean() ?: mapConfiguration[param]!!
+    }
+
+    companion object {
+        val ignoreNumbersList = listOf("-1", "1", "0", "2")
```

Comment:
We should parse these numbers as numbers when creating configuration instance

Reply:
But there can be a long(`1L`) or float(`1.0f`) type, which we should also ignore. I thought to store all types of numbers in one list as String.

Reply:
Won't they have a different node element type (like integer constant vs double constant or something)? Also, you are then comparing these numbers by text, does it work then? Do you have tests for `1.0f`?

Reply:
Yes, there are two types: `INTEGER_CONSTANT` and `FLOAT_CONSTANT` and I check it in `logic()`. I a little changed logic, but yes, I had test for this.